### DISCUSSION TITLE
Validate and build graph

### DISCUSSION THREAD

**Discussion by redlars** [microsoft/vcpkg#17786] [Ideas]
Using latest version of vcpkg on Windows 10 Pro.

Two questions;
1. The command "vcpkg install gdal" needed to install 101 packages (which is fine). It ran for a LONG time but failed on package 93 because Oracle has dropped support for libmysql for 32-bit (which is also fine). It would have been nice to get this error earlier. Obvioulsy do not know the inner workings of vcpkg but a pre-validation step for all packages would be nice to catch any obvious issues.

2. It would be very helpful if vcpkg could output a "building graph" of a given package. The command "vcpkg install gdal" will list all 101 packages in a list but does not say anything about the relationship between each package. It does not tell me which package requires "boost-Phoenix".

**Comment by past-due** [comment]
I second **2** as a very useful feature addition.

There's obviously a limit to what can be done for **1**, but a perhaps a "target-support" attribute in the `CONTROL` file that could specify target and/or architecture combinations?

So instead of having (for example) [a check for "WindowsStore" inside the portfile.cmake](https://github.com/Microsoft/vcpkg/blob/79a526a3753553fd9b1f874ada7688b2fb8396a4/ports/glib/portfile.cmake#L2) for a port that doesn't yet support WindowsStore builds, the `CONTROL` file could specify:
`Target-Support: !WindowsStore`
(which would exclude WindowsStore / UWP builds, but implicitly allow anything else)

If a port doesn't support x86 / 32-bit builds, the `CONTROL` file could specify:
`Target-Support: !x86`
or
`Target-Support: !32bit` (which would exclude _any_ 32-bit target)

Or, if a port doesn't support x86 Windows builds (but does support x64 Windows builds):
`Target-Support: !Windows[x86]`

Or for a port that supports all builds (including 64-bit macOS builds) but *not* 32-bit macOS builds:
`Target-Support: !macOS[x86]`

Alternatively, name it `Target-Unsupported` and omit the `!`?

**Comment by JackBoosY** [comment]
Hmm... I think this is a good suggestion, and we already added keyword `Support` in CONTROL.
Reserved.
